### DISCUSSION TITLE
Ideas for renaming `grossDepositAmount`

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#304] [Ideas]
As discussed in #204, I find the `grossDepositAmount` param name rather verbose. It's a good fit for the create functions but there's something bugging me about there being three words in the name. In addition, there's a question of marketing at play here - we don't want to give an indication that fees are charged by default when in fact they aren't.

Suggested alternatives ordered by how much I like them:

1. `totalAmount`
2. `depositAmount`
3. `lockupAmount`
4. `grossAmount`
5. `amount`

None is great, and I can come up with an explanation for why `grossDepositAmount` is a better fit than each of the above. But we gotta start somewhere - WDYT?

For me, `depositAmount` is the leading candidate, since I expect many V2 streams to be fee-less. For fee-less streams, the "gross deposit amount" is the same as the struct "deposit amount". Conversely, the fees should be small anyway, so it's fine to call the function name "deposit amount" just like the struct because they will have approximately the same value.

`totalAmount` is my second favorite candidate, because it communicates the fact that there is something more than just the deposit amount in this function param (i.e. the amount that gets stored in the struct as `.amounts.deposit`).

Alternatively, we could consider renaming the `amounts.deposit` in the struct to something else .. though that would also mean renaming `getDepositAmount`.

**Comment by andreivladbrg** [comment]
I don't think that using the "deposit" word is accurate because:

```
grossDepositAmount = protocolFee + brokerFee + netDeposit;
```

And the `brokerFee` goes from  funder --> broker.addr, it means that is not actually deposited. 

My choices are:
 - totalAmount
 - grossAmount

**Reply by PaulRBerg** [reply]
Excellent point - I didn't even consider the fact that the broker fee is not wired through the Sablier contracts.

Through this lens, it does indeed make more sense to use something like `totalAmount` or `grossAmount`.

I tend to like the former better because "total" is a simpler word than "gross".

**Comment by PaulRBerg** [comment]
Closing as #305 has been merged.

We ended up going with `totalAmount`.
